### PR TITLE
[DPO] fix DPO ref_model=None

### LINKED ISSUE #699: Unable to use `ref_model=None` with DPOTrainer
Hi,

While trying to run DPO with `ref_model=None` ([PR #640](https://github.com/huggingface/trl/pull/640)), I ran into a couple of issues.

Following the Stack-Llama-2 example, `dpo_llama2.py` runs ok with all the default settings. However, this example loads the active and ref model separately and passes both to the `DPOTrainer`. 

If I understand correctly, setting `ref_model=None`, should avoid copying the model and instead disable the adapters to compute the reference model score. But it seems like [this check](https://github.com/huggingface/trl/blob/d3bbee3ab84d4d5dcb413003bf93541df5b4e54c/trl/trainer/dpo_trainer.py#L116) consistently fails and the model gets copied in any case. This also produces the following warning with no updates to the loss:

```python
 UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
```

To reproduce, replace https://github.com/huggingface/trl/blob/d3bbee3ab84d4d5dcb413003bf93541df5b4e54c/examples/research_projects/stack_llama_2/scripts/dpo_llama2.py#L141C5-L146C6 with `model_ref=None`.

---

Libraries:

```
accelerate               0.21.0
bitsandbytes             0.41.1
peft                     0.5.0
torch                    2.0.1
torchvision              0.15.2
transformers             4.33.0.dev0
trl                      0.6.1.dev0
```

Any insights on this would be greatly appreciated!

### PR DESCRIPTION
**Pull request by kashif**
by @tannonk fixes #699  and #701

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thank you for the quick fix @kashif !

File: trl/trainer/dpo_trainer.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def __init__(
                 model = prepare_model_for_int8_training(model)
             model = get_peft_model(model, peft_config)
 
-        self.is_peft_model = getattr(model, "is_peft_model", False)
+        self.is_peft_model = getattr(model, "active_adapter", False)
```
Comment (younesbelkada) [inline trl/trainer/dpo_trainer.py:116]: nit, this can be also:

```suggestion
        self.is_peft_model = is_peft_available() and isinstance(model, PeftModel)
```
